x-pack/plugins/data_visualizer/public/plugin.ts

```diff
@@ -50,7 +52,15 @@ export class DataVisualizerPlugin
     }
   }
 
-  public setup(core: DataVisualizerCoreSetup, plugins: DataVisualizerSetupDependencies) {
+  public async setup(core: DataVisualizerCoreSetup, plugins: DataVisualizerSetupDependencies) {
+    const [coreStart, pluginStart] = await core.getStartServices();
```

Comment:
registerEmbeddables is still occuring after the async task. Since registerEmbeddables does not use coreStart or pluginStart, move registerEmbeddables to before `await`

Reply:
Updated here https://github.com/elastic/kibana/pull/184030/commits/0a39eb7364e235191c4f416dc3881afa8ef510e0

Would also be good to document that registerEmbeddables should not be after an async task. I see this registerEmbeddables being called in several places after an await in the code base.